railgun/hack/generators/controllers/networking/main.go

```diff
@@ -234,14 +267,66 @@ var controllerTemplate = `
 type {{.PackageAlias}}{{.Type}}Reconciler struct {
 	client.Client
 
+{{- if or .AcceptsIngressClassNameSpec .AcceptsIngressClassNameAnnotation}}
+	Log              logr.Logger
+	Scheme           *runtime.Scheme
+	Proxy            proxy.Proxy
+	IngressClassName string
+{{- else}}
 	Log    logr.Logger
 	Scheme *runtime.Scheme
 	Proxy  proxy.Proxy
+{{- end}}
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *{{.PackageAlias}}{{.Type}}Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+{{- if .AcceptsIngressClassNameAnnotation}}
+	preds := predicate.NewPredicateFuncs(func(obj client.Object) bool {
+		if v, ok := obj.GetAnnotations()["kubernetes.io/ingress.class"]; ok {
```

Comment:
the `"kubernetes.io/ingress.class"` magic string already exists as a const: `annotations.IngressClassKey`

Reply:
acbe0bcaccf4231189496796b60ecdf9cd97f291 resolves this (across the entire code base)